### PR TITLE
Firefly-1620: more spectrum types for the guesser

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1620](https://jira.ipac.caltech.edu/browse/FIREFLY-1620): more spectrum types for the guesser

- recognize spectrum from next-sci
- spectrum plot will reverse the low spectrum if negative

#### Testing
- https://fireflydev.ipac.caltech.edu/firefly-1620-spec-types/firefly
- upload some of the files in the ticket (make sure to check `Attempt to interpret tables as spectra`)

### CODE REVIEW

File: src/firefly/js/charts/dataTypes/FireflySpectrum.js [hunk 1]
```diff
@@ -131,7 +132,18 @@ export function getSpectrumProps(tbl_id, spectrumDM) {
...
     const xErrArrayMinus = spectralAxis.statErrLow;
     const yErrArray = fluxAxis.statErrHigh || fluxAxis.statError;
-    const yErrArrayMinus = fluxAxis.statErrLow;
+    let yErrArrayMinus = fluxAxis.statErrLow;
+    if (yErrArray && fluxAxis.statErrLow) {
+        const errCol= getColumnIdx(tbl,yErrArray);
+        const lowCol= getColumnIdx(tbl,fluxAxis.statErrLow);
+        let allEmpty=true;
+        const reverse= tbl.tableData.data.every( (r) => {
+            if (isNaN(r[errCol]) || isNaN(r[lowCol]) || r[errCol===null] || r[lowCol]===null) return true;
```
Comment (jaladh-singhal) [inline src/firefly/js/charts/dataTypes/FireflySpectrum.js:141]:
```suggestion
            if (isNaN(r[errCol]) || isNaN(r[lowCol]) || r[errCol]===null || r[lowCol]===null) return true;
```
I think you meant this?

**Review by jaladh-singhal — APPROVED** [review]
Looks good except there's a typo I think

Also, when I upload 2MASS file, all error values are null but I still see the error bars (+), is that expected?

File: src/firefly/js/charts/dataTypes/FireflySpectrum.js (hunk 1 above)
Comment (robyww) [reply src/firefly/js/charts/dataTypes/FireflySpectrum.js:141]: yes, I will fix it.

**Comment by robyww** [comment]
> Also, when I upload 2MASS file, all error values are null but I still see the error bars (+), is that expected?

I think that is the way it is suppose to work. I am not totally sure but it makes sense. The upper and lower errors are effectively 0

**Pull request merged**